### PR TITLE
[16.0][MIG] account_invoice_inter_company: Migration to 16.0

### PR DESCRIPTION
**Pull request by chafique-delli**
cc @sebastienbeau, @Kev-Roche

### CODE REVIEW

**Comment by moshchot** [comment]
hi @chafique-delli  we're testing your PR and however we try, on runboat and on our dev instance, odooBot stopped being an option for intercompany user for auto validation.
is that  deliberate? if so, should we create a superuser of one own? 

image

**Comment by chafique-delli** [comment]
> hi @chafique-delli we're testing your PR and however we try, on runboat and on our dev instance, odooBot stopped being an option for intercompany user for auto validation. is that deliberate? if so, should we create a superuser of one own?
> 
> image

hi @moshchot , 
There are no restrictions on the choice of the intercompany user.
There is no obligation to choose an intercompany user.
If you choose one, he must have the rights to create invoices.

File: account_invoice_inter_company/views/res_config_settings_view.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                        title="Values set here are company-specific."
+                        aria-label="Values set here are company-specific."
+                        groups="base.group_multi_company"
+                        role="img"
+                    />
+                </div>
+            </xpath>
+            <xpath expr="//div[@id='inter_company']/div[2]" position="inside">
+                <div id="company_share_product">
+                    <field name="company_share_product" class="oe_inline" />
```
Comment (Kev-Roche) [inline account_invoice_inter_company/views/res_config_settings_view.xml:23]: The help in view is not correct

File: account_invoice_inter_company/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "name": "Inter Company Invoices",
+    "summary": "Intercompany invoice rules",
+    "version": "16.0.1.0.0",
+    "category": "Accounting & Finance",
+    "website": "https://github.com/OCA/multi-company",
+    "author": "Odoo SA, Akretion, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "depends": ["account", "base_setup"],
+    "data": [
+        "data/inter_company_data.xml",
```
Comment (Kev-Roche) [inline account_invoice_inter_company/__manifest__.py]: I think it s demo, not data.

**Review by Kev-Roche — CHANGES_REQUESTED** [review]

File: account_invoice_inter_company/data/inter_company_data.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <!--
+        Create inter company product
+        -->
+    <record id="product_consultant_multi_company" model="product.product">
+        <field name="name">Service Multi Company</field>
+        <field name="uom_id" ref="uom.product_uom_hour" />
+        <field name="uom_po_id" ref="uom.product_uom_hour" />
+        <field name="categ_id" ref="product.product_category_3" />
```
Comment (hbrunn) [inline account_invoice_inter_company/data/inter_company_data.xml]: this one is from [demo data](https://github.com/OCA/OCB/blob/16.0/addons/product/data/product_demo.xml#L13), should the whole thing then be demo?

**Review by hbrunn — CHANGES_REQUESTED** [review]

File: account_invoice_inter_company/models/account_move.py [hunk 4]
```diff
@@ -154,6 +157,25 @@ def _inter_company_create_invoice(self, dest_company):
...
+                        "The invoice line '%(line_name)s' doesn't have a product. "
+                        "All invoice lines should have a product for "
+                        "inter-company invoices."
+                    )
+                    % {"line_name": src_line.name}
+                )
+            dest_move_line_data.append(
+                src_line._prepare_account_move_line(dest_invoice, dest_company)
+            )
+            self.env["account.move.line"].create(dest_move_line_data)
```
Comment (hbrunn) [inline account_invoice_inter_company/models/account_move.py]: this should go out of the loop, right?

File: account_invoice_inter_company/__manifest__.py (hunk 2 above)
Comment (chafique-delli) [reply account_invoice_inter_company/__manifest__.py]: @Kev-Roche , @hbrunn , Indeed, I had asked myself the question.

File: account_invoice_inter_company/views/res_config_settings_view.xml (hunk 1 above)
Comment (chafique-delli) [reply account_invoice_inter_company/views/res_config_settings_view.xml:23]: @Kev-Roche , the help came from the base_setup module, I removed it.

File: account_invoice_inter_company/__manifest__.py (hunk 2 above)
Comment (chafique-delli) [reply account_invoice_inter_company/__manifest__.py]: @Kev-Roche , @hbrunn , it's modified.

File: account_invoice_inter_company/data/inter_company_data.xml (hunk 3 above)
Comment (chafique-delli) [reply account_invoice_inter_company/data/inter_company_data.xml]: @hbrunn , it's done.

File: account_invoice_inter_company/models/account_move.py (hunk 4 above)
Comment (chafique-delli) [reply account_invoice_inter_company/models/account_move.py]: @hbrunn , it's right. It's fixed.

**Review by hbrunn — APPROVED** [review]

**Review by Kev-Roche — APPROVED** [review]

**Pull request merged**